src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -880,7 +913,34 @@ void QgsRelationReferenceWidget::filterChanged()
 
 void QgsRelationReferenceWidget::addEntry()
 {
-  QgsFeature f( mReferencedLayer->fields() );
+  if ( !mReferencedLayer )
+    return;
+
+  const QgsVectorLayerTools *tools = mEditorContext.vectorLayerTools();
+  if ( !tools )
+    return;
+  if ( !mCanvas )
+    return;
+
+  mMapToolDigitize->setLayer( mReferencedLayer );
+  setMapTool( mMapToolDigitize );
+
+  connect( mMapToolDigitize, &QgsMapToolDigitizeFeature::digitizingCompleted, this, &QgsRelationReferenceWidget::entryAdded );
+  connect( mCanvas, &QgsMapCanvas::keyPressed, this, &QgsRelationReferenceWidget::onKeyPressed );
+
+  if ( mMessageBar )
+  {
+    QString title = tr( "Relation %1 for %2." ).arg( mRelationName, mReferencingLayer->name() );
```

Comment:
Would the field name be more interesting here? And possibly the displayString of the edited feature?
I'm imagining something like

> *Link feature to layer manhole "XC123"* : Digitize the geometry for the new feature on layer cover. Press Esc to cancel.

Reply:
I change the message but the feature display string is way more difficult to add. Indeed, I assume you talk about the **Column preview** that change the main display of feature list.

The relation reference widget has no reference on this column preview expression. We would need to add a variable in QgsAttributeEditorContext and set it from the QgsAttributeForm or something like that. Do you think it worth the modification only for this message to be displayed?

Reply:
Not sure what the column preview is, the main display uses the displayString of the layer if I'm not wrong.

Reply:
Something like this 89e76ca ? 

I add a setFormFeature method on the qgsrelationreferencewidget, and call the method from the wrapper. Is that OK to do this ?

Reply:
Yes, that should be ok.
Just one small thing, prefer putting the method definition into the cpp file.

Reply:
OK, done